### ISSUE TITLE
nan loss while training

### ISSUE THREAD

**Issue by DebasmitaGhose** [jwyang/faster-rcnn.pytorch#349]
I have been trying to train a faster r-cnn network on another dataset and have run into issues with nan training loss. I have been trying to track down the nan loss and it seems to be coming from the `bbox_tranform_batch` function in `bbox_transform.py`. 
Specifically, I have noticed, that these two are the first to become nan when the training begins
```
 targets_dw = torch.log(gt_widths / ex_widths)
 targets_dh = torch.log(gt_heights / ex_heights)

```
Can somebody think of a possible reason as to how this can fixed?

**Comment by rnjtsh** [comment]
I faced the same issue while training on my own dataset which is similar to pascal voc. You need to modify these four [lines](https://github.com/jwyang/faster-rcnn.pytorch/blob/master/lib/datasets/pascal_voc.py#L234-L237) while reading the ground truth according to your dataset. Check whether you need to remove or keep the +1.

**Comment by zhengxinvip** [comment]
> Hi, I printed the values for gt_widths, gt_heights, ex_widths and ex_heights. None of them are zeros. Could it be a problem with torch.log described here
> Edit: I tried experimenting with different images and I noticed that sometimes, all values in the tensor ex_widths are positive and all values in the tensor gt_widths are negative, which means that the log of negative values is being calculated. Any idea on why this could be happening?
> What are the correct signs of gt_widths and ex_widths?

Did you solve the problem. I meet the same error. I need your help . thanks.

**Comment by DebasmitaGhose** [comment]
@zhengxinvip we did solve the problem. 
Turns out, we were reading our annotations wrong in our code. 
This code needs to have the coordinates of the top left and bottom right corners of the bounding boxes, so you might have to compensate for that if your annotations are not in that format. 
Our annotations were in the form of  [l t w h]  and [this](https://github.com/DebasmitaGhose/Multimodal_Influenza_Detection/blob/fppi_LAMR/faster-rcnn.pytorch/lib/datasets/kaist_fusion.py#L243) is how we solved our problem.

**Comment by marcunzueta** [comment]
check new comment from #594

**Comment by HassanAlsamahi** [comment]
> I have been trying to train a faster r-cnn network on another dataset and have run into issues with nan training loss. I have been trying to track down the nan loss and it seems to be coming from the `bbox_tranform_batch` function in `bbox_transform.py`.
> Specifically, I have noticed, that these two are the first to become nan when the training begins
> 
> ```
>  targets_dw = torch.log(gt_widths / ex_widths)
>  targets_dh = torch.log(gt_heights / ex_heights)
> ```
> 
> Can somebody think of a possible reason as to how this can fixed?

how did you track down the nan loss??
